### PR TITLE
add definition on conditions usage

### PR DESCRIPTION
**Pull request by stuggi**

### CODE REVIEW

File: conditions.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+**Note**: While it might be useful/easier to manage conditions in a map (easier to access/set) instead of a list, conditions are represented using a list/slice of objects, where each condition has a similar structure. This collection should be treated as a map with a key of type.
+
+## Conditions in openstack-k8s-operators podified controlplane operators
+* At the beginning of the reconciliation loop, controller adds a `Ready` condition which reflects the overall state of the service with `Status=unknown`
+* When a controller performs a task, before it starts a step condition gets initialized to `Status=unknown` e.g. DBReady, DBSync, Endpoint, ServiceRegistered, …
+* At the end of the reconcile loop the overall `Ready` condition gets set to `Status=True/False`, depending if the sub task conditions are fulfilled or not.
+* The service `Ready` condition with `Status=True` reflects that the service is configured, up and accessible.
+* If the `Ready` condition is `Status=False` or `Unknown` the service is not ok to answer requests
```
Comment (gibizer) [inline conditions.md:14]: nit: OK

File: conditions.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  - `Status=False` indicates an Error that needs user intervention to fix it
+  - `Status=unknown` indicates that the operator can progress without user intervention. i.e the operator is waiting for a job to complete
```
Comment (gibizer) [inline conditions.md:16]: nit: Unknown

File: conditions.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
+# Conditions
+This doc describes how condiditons should be used in operators managing a podified controlplane (**NOT** osp-director-operator).
+
+## General information
+Description of conditions https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties 
+
+**Note**: While it might be useful/easier to manage conditions in a map (easier to access/set) instead of a list, conditions are represented using a list/slice of objects, where each condition has a similar structure. This collection should be treated as a map with a key of type.
+
+## Conditions in openstack-k8s-operators podified controlplane operators
+* At the beginning of the reconciliation loop, controller adds a `Ready` condition which reflects the overall state of the service with `Status=unknown`
```
Comment (gibizer) [inline conditions.md:10]: nit: Unknown

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**